encoding/fft/fft_g1.go

```diff
@@ -107,9 +132,19 @@ func (fs *FFTSettings) FFTG1(vals []bn254.G1Affine, inv bool) ([]bn254.G1Affine,
 	// We make a copy so we can mutate it during the work.
 	valsCopy := make([]bn254.G1Affine, n)
 	for i := 0; i < len(vals); i++ { // TODO: maybe optimize this away, and write back to original input array?
-
 		valsCopy[i].Set(&vals[i])
 	}
+
+	// _fftG1 will spawn goroutines until maxSplits is reached,
+	// effectively spawning nextPowOf2(numCPU) goroutines at most.
+	// every node of the recursion tree up to maxSplits spawns a goroutine for 1/2 of the work.
+	// Since there are 2*2^maxSplits nodes in the tree, this will lead to 2^maxSplits goroutines.
+	// Ultimately, this means each leaf at depth maxSplits is run concurrently in a goroutine.
+	// Surprisingly, increasing maxSplits way past numCPU improves performance (slightly)...
```

Comment:
Is the surprising part that increasing the maxSplits past numCPU improves the performance only slightly or that it improves the performance at all?

Reply:
At all. Would have assumed if you create 2^20 goroutines it should actually be slower, but it’s not; it’s faster.